tests/integration/test_main.py

```diff
@@ -66,13 +66,52 @@ def test_run():
         run()
 
 
-def test_empty_run(tmp_path, monkeypatch):
+def test_empty_run(tmp_path):
     """Test real run with no diags."""
-    monkeypatch.delitem(  # TODO: remove in v2.14.0
-        esmvalcore.config.CFG._mapping,
-        "config_file",
-        raising=False,
+    recipe_file = tmp_path / "recipe.yml"
+    content = dedent("""
+        documentation:
+          title: Test recipe
+          description: This is a test recipe.
+          authors:
+            - andela_bouwe
+          references:
+            - contact_authors
+            - acknow_project
+          projects:
+            - c3s-magic
+        diagnostics: null
+    """)
+    recipe_file.write_text(content)
+    log_dir = f"{tmp_path}/esmvaltool_output"
+    config_dir = tmp_path / "config"
+    config_dir.mkdir(parents=True, exist_ok=True)
+    config_file = config_dir / "config.yml"
+    config_file.write_text(f"output_dir: {log_dir}")
+
+    msg = "The given recipe does not have any diagnostic."
+    with warnings.catch_warnings():
+        warnings.simplefilter("error")
+        with pytest.raises(RecipeError, match=msg):
+            ESMValTool().run(recipe_file, config_dir=config_dir)
+    log_file = os.path.join(
+        log_dir,
+        os.listdir(log_dir)[0],
+        "run",
+        "main_log.txt",
     )
+    filled_recipe = os.path.exists(
+        log_dir + "/" + os.listdir(log_dir)[0] + "/run/recipe_filled.yml",
```

Comment:
Would it be worth using `os.path.join` here too?

Reply:
Good point! I think the logic in that test was also flawed (it called `shutil.rmtree` and then asserts that a file does not exist within the removed dir?). Fixed that in https://github.com/ESMValGroup/ESMValCore/pull/2736/commits/078b8b303a507397afb46f90d212d8665529520f